### PR TITLE
Sql/blob auditing

### PR DESCRIPTION
**Pull request by marstr**
This PR has been created by me for @bashahee.

The changes through 2d07b89 in this PR have previously been evaluated in the following PRs:

- Azure/azure-rest-api-specs-pr#342
- Azure/azure-rest-api-specs-pr#421

Any subsequent commits to work around conflicts will be subject to further review.

### CODE REVIEW

**Comment by marstr** [comment]
@bashahee in order to make modifcations to this PR, you can submit changes to the sql/blobAuditing branch. In the case that you do not have push permissions to that branch, you can submit PRs against it, and request a review from me.

**Comment by jaredmoo** [comment]
Thanks for notifying me, I will make sure to review this.

**Comment by hovsepm** [comment]
Please fix mere conflicts and build breaks.

**Review by jaredmoo — CHANGES_REQUESTED** [review]

**Comment by marstr** [comment]
@bashahee (and any other interested parties viewing this thread), we've updated our policies around who is allowed to push to unprotected branches. You can read about it [here](https://github.com/Azure/azure-rest-api-specs/wiki#branch-strategy).

What it comes down to, is that if you join the [GitHub team azure-rest-api-writers](https://repos.opensource.microsoft.com/azure/teams/azure-rest-api-writers), you'll be able to contribute to this PR directly instead of going through an intermediate PR.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json]: Confirmed.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json]: Yes,  DB (non-extended) policy also in 2017-03-01.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: @jaredmoo Old one should be removed.

Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: Ok, will you remove it?

File: specification/sql/resource-manager/readme.md
Comment (jaredmoo) [inline specification/sql/resource-manager/readme.md]: Please delete this and add to `package-pure-2017-03-preview` instead

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json
Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json]: Cool. Is it backwards compatible with DB policy in 2015-05-01?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: Add this file to readme.md in package-pure-2015-05-preview

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/blobAuditing.json]: Yes.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: @jaredmoo Done.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/blobAuditing.json]: @jaredmoo Done.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/operations.json
Comment (bashahee) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/operations.json]: @jaredmoo Done

File: specification/sql/resource-manager/readme.md
Comment (bashahee) [reply specification/sql/resource-manager/readme.md]: Done.

**Review by jaredmoo — APPROVED** [review]

**Comment by jaredmoo** [comment]
Please do not merge until AFTER https://github.com/Azure/azure-sdk-for-net/pull/4351 is merged. Thx

**Comment by hovsepm** [comment]
@bashahee please fix build breaks. Also please ping me when this PR can be merged.

**Comment by jaredmoo** [comment]
The SDK change is merged, but now we need https://github.com/Azure/azure-rest-api-specs/pull/3134 merged and then please change this one to use the common definitions. Thx

**Comment by jaredmoo** [comment]
Python build failure due to app insights:

```
INFO:swaggertosdk.autorest_tools:ERROR (Fatal/DuplicateModelCollsion): Duplicated model name with non-identical definitions
INFO:swaggertosdk.autorest_tools:    - file:///git-restapi/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/workbooks_API.json:258:4 ($.definitions.Resource)
INFO:swaggertosdk.autorest_tools:    - file:///git-restapi/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/webTests_API.json:285:4 ($.definitions.Resource)
INFO:swaggertosdk.autorest_tools:    - file:///git-restapi/specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json:292:4 ($.definitions.Resource)
```

**Comment by bashahee** [comment]
@hovsepm This PR can be merged.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseBlobAuditingCreateMax.json
Comment (hovsepm) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseBlobAuditingCreateMax.json]: >  "name": "default", [](start = 6, length = 20)

Does your service always return `default` value for Name parameter in response to PUT operation?

File: specification/sql/resource-manager/readme.md
Comment (hovsepm) [inline specification/sql/resource-manager/readme.md]: Are you sure that tag `package-2017-03-preview` should contain 2 versions of `blobAuditing.json`? 2015-05-01-preview and 2017-03-01-preview?

**Comment by hovsepm** [comment]
@fearthecowboy could you help me understand the merging error in linter validation?
```
(node:7221) UnhandledPromiseRejectionWarning: Error: '$.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/auditingSettings/{blobAuditingPolicyName}"].get.parameters' has incompatible values (---
- $ref: '#/parameters/ResourceGroupParameter'
- $ref: '#/parameters/ServerNameParameter'
- $ref: '#/parameters/DatabaseNameParameter'
- $ref: '#/parameters/BlobAuditingPolicyNameParameter'
- $ref: '#/parameters/SubscriptionIdParameter'
- name: api-version
  in: query
  description: The API version to use for the request.
  required: true
  type: string
  x-ms-parameter-location: client
  enum:
    - 2015-05-01-preview
, ---
- $ref: '#/parameters/ResourceGroupParameter'
- $ref: '#/parameters/ServerNameParameter'
- $ref: '#/parameters/DatabaseNameParameter'
- $ref: '#/parameters/BlobAuditingPolicyNameParameter'
- $ref: '#/parameters/SubscriptionIdParameter'
- name: api-version
  in: query
  description: The API version to use for the request.
  required: true
  type: string
  x-ms-parameter-location: client
  enum:
    - 2017-03-01-preview
).
    at Merge (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/source-map/merging.js:57:11)
    at Merge (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/source-map/merging.js:52:31)
    at Merge (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/source-map/merging.js:52:31)
    at Merge (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/source-map/merging.js:52:31)
    at Merge (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/source-map/merging.js:52:31)
    at Object.MergeYamls (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/source-map/merging.js:193:24)
    at Object.ComposeSwaggers (/home/travis/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist/lib/pipeline/swagger-loader.js:368:36)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```
I'm not clear why there is an API version conflict. is it becaouse of the `package-2017-03-preview` tag in readme.md that contains two different API versions of the same swagger? (blobAuditing.json 2015-05-01-preview and 2017-03-01-preview)

**Comment by jaredmoo** [comment]
Looks like package-2017-03-preview should remove blobAuditing.json 2015-05-01-preview.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseBlobAuditingCreateMax.json
Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/examples/DatabaseBlobAuditingCreateMax.json]: This is a "singleton resource", so "default" is the only valid name.

**Comment by bashahee** [comment]
@hovsepm Fixed merge errors, build warnings are not related to blob auditing.

**Comment by bashahee** [comment]
@hovsepm Ready to be merged.

**Comment by hovsepm** [comment]
@bashahee there are bunch of changed files now that are not in the specification/sql folder (e,g, automation, applicationinsights) - please rebase your code and exclude files in other teams folder from your PR.

**Comment by bashahee** [comment]
@hovsepm Without these changes, there will be build errors. I basically did what was done in PR #3134 .

**Comment by hovsepm** [comment]
@marstr could you please take a look to this PR? In particular I do not have clear idea on why there are bunch of changed files that are not in the specification/sql folder (e,g, automation, applicationinsights). I cannot approve a PR to SQL that affects other service swaggers without:
1. Clear sign-off from the affected swagger owners
2. Proper justification (there should be no build error on other teams swaggers or otherwise our build infrastructure has bugs).

**Comment by jaredmoo** [comment]
@bashahee , can you undo the changes to RPs other than SQL? It should not be our concern if their build fails.

**Comment by bashahee** [comment]
@jaredmoo @hovsepm 
If I undo changes to RPs other than SQL, there will be build errors. In particular, "C:\Code\azure-rest-api-specs\specification\automation\resource-manager>autorest readme.md --multiapi --python --python-mode=update --python-sdks-folder=/tmp/tmpt0dy8527/sdk --use=@microsoft.azure/autorest.python@~3.0 --version=preview" fails for other "readme.md" files.

**Comment by jaredmoo** [comment]
@bashahee I think we should let @lmazuel handle that problem with the other RP's. We should not change other RP's swaggers for any reason.

**Comment by bashahee** [comment]
@hovsepm The PR should be clean from unrelated changes now. Please merge it if you see so.

**Review by hovsepm — APPROVED** [review]

**Pull request merged**
